src/lib/hooks/useGetMaxLengthError.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback } from "react";
+
+import { useCelatoneApp } from "lib/app-provider";
+
+export const useGetMaxLengthError = () => {
+  const { constants } = useCelatoneApp();
+  return useCallback(
+    (
+      field: string,
+      currentLength: number,
+      type:
+        | "contract_name"
+        | "contract_desc"
+        | "code_name"
+        | "proposal_title"
+        | "list_name"
+    ) => {
+      let maxLength = 0;
+      switch (type) {
+        case "contract_name":
+          maxLength = constants.maxContractNameLength;
+          break;
+        case "contract_desc":
+          maxLength = constants.maxContractDescriptionLength;
+          break;
+        case "code_name":
+          maxLength = constants.maxCodeNameLength;
+          break;
+        case "proposal_title":
+          maxLength = constants.maxProposalTitleLength;
+          break;
+        case "list_name":
+          maxLength = constants.maxListNameLength;
+          break;
+        default:
+          break;
+      }
+      return `${field} is too long. (${currentLength}/${maxLength})`;
```

Comment:
I recommend that you do the following:
```tsx
import { useCallback } from "react";

import { useCelatoneApp } from "lib/app-provider";

type MaxLengthType =
  | "contract_name"
  | "contract_desc"
  | "code_name"
  | "proposal_title"
  | "list_name";

export const useGetMaxLengthError = () => {
  const { constants } = useCelatoneApp();
  return useCallback(
    (currentLength: number, type: MaxLengthVariant) => {
      const maxLengthMap: Record<MaxLengthType, number> = {
        contract_name: constants.maxContractNameLength,
        contract_desc: constants.maxContractDescriptionLength,
        code_name: constants.maxCodeNameLength,
        proposal_title: constants.maxProposalTitleLength,
        list_name: constants.maxListNameLength,
      };

      const fieldMap: Record<MaxLengthType, string> = {
        contract_name: "Contract name",
        contract_desc: "Contract description",
        code_name: "Code name",
        proposal_title: "Proposal title",
        list_name: "List name",
      };

      const maxLength = maxLengthMap[type] ?? 0;
      const fieldName = fieldMap[type] ?? "Unknown field";

      return `${fieldName} is too long. (${currentLength}/${maxLength})`;
    },
    [
      constants.maxCodeNameLength,
      constants.maxContractDescriptionLength,
      constants.maxContractNameLength,
      constants.maxListNameLength,
      constants.maxProposalTitleLength,
    ]
  );
};
```
and simply supplying 'length' and 'type' as parameters to this function.